bdai_ros2_wrappers/bdai_ros2_wrappers/service.py

```diff
@@ -31,7 +31,76 @@ class ServiceError(ServiceException):
     pass
 
 
-class Serviced(ComposableCallable, VectorizingCallable):
+ServiceRequestT = TypeVar("ServiceRequestT", contravariant=True)
+ServiceResponseT = TypeVar("ServiceResponseT", covariant=True)
+
+
+class ServicedProtocol(Protocol[ServiceRequestT, ServiceResponseT]):
```

Comment:
I don't like having to split types like this, but nested types on type parameters are too much for `mypy` to handle.